modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java

```diff
@@ -346,25 +363,27 @@ private void parseWsdl() throws InitialisationException
     {
         Definition wsdlDefinition = null;
 
+        InputStream inputStream = null;
+        
+
         URL url = IOUtils.getResourceAsUrl(config.getWsdlLocation(), getClass());
         if (url == null)
         {
             throw new InitialisationException(MessageFactory.createStaticMessage("Can't find wsdl at %s", config.getWsdlLocation()), this);
         }
 
+        try {
+            inputStream = wsdlRetrieverStrategy.retrieveWsdl(url);
+        } catch (Exception e) {
+            throw new InitialisationException(MessageFactory.createStaticMessage("Couldn't retrieve wsdl at %s", config.getWsdlLocation()), this);
+        }
+
         try
         {
             WSDLReader wsdlReader = WSDLFactory.newInstance().newWSDLReader();
-
-            URLConnection urlConnection = url.openConnection();
-            if (url.getUserInfo() != null)
-            {
-                urlConnection.setRequestProperty("Authorization", "Basic " + Base64.encodeBytes(url.getUserInfo().getBytes()));
-            }
-
-            wsdlDefinition = wsdlReader.readWSDL(url.toString(), new InputSource(urlConnection.getInputStream()));
+            wsdlDefinition = wsdlReader.readWSDL(url.toString(), new InputSource(inputStream));
```

Comment:
According to the last commit, the strategy is the class responsible for closing the stream as well as the httpclient (it returns a Wsdl Definition). So this should review there.